### PR TITLE
Skin HTML components like the Swing ones

### PR DESCRIPTION
**Pull request by JaroslavTulach**
This PR contains several bugfixes and a major initial step forward - any HTML component created by one of `@HTMLComponent`, `@HTMLDialog`, etc. annotations will get proper CSS according to currently active Swing Look&Feel.

The styling is based on [work done](https://github.com/emilianbold/netbeans-releases/commit/cf35ecd4b7284572f9d7ab7e223781d3c69cbf15#diff-382d068b6686b32b5e3886e8cf0bf9df) by Toni Epple. Plus I've added a simple **darcula** support.

The work isn't finished, but it is certainly nicer than having nothing. It would be great if more CSS experts could chim in and improve the CSS for more HTML elements - however that is out of scope of my change.

### CODE REVIEW

**Comment by Chris2011** [comment]
Can you make a little example or do you have a documentation for this? Because I created a plugin called NbScratchFile and I use javaFX webview w/o any Html4J and w/o dukescript, it caused problems on my side, so I created it by my self.
But maybe your approach is better and give some advantages and maybe a better performance?

https://github.com/Chris2011/NbScratchFile/blob/develop/src/main/resources/org/chrisle/netbeans/plugins/utils/WebViewDialog.java

Let us talk separately, which channel do you prefer?

**Comment by JaroslavTulach** [comment]
Documentation of these annotations for version 8.2 is [available here](http://bits.netbeans.org/8.2/javadoc/org-netbeans-api-htmlui/overview-summary.html). There are even wizards in the IDE that generate the HTML based UI (component, dialog, as well as wizard for your file types). In general I believe in shared infrastructure rather than doing the same thing again and again from scratch. When people learn how to join forces they usually achieve more than on their own.

**Comment by Chris2011** [comment]
Sure me too, but as I said, I had some problems while using dukescript and there is no issue tracker to let them know it. Anyway, thx for the info, I will try to refactor my code to use the HtmlUI 👍. But for me as a FrontEnd WebDeveloper, you can see it in my code, what my use case is. No mixed JS/Java code. Clean HtmlUI with own build process and JS/TS, decoupled from the other logic. Sry for spamming into this PR. I'm finished now :')

**Comment by geertjanw** [comment]
There's no bug tracker? And who is 'them'? @JaroslavTulach is one of the key developers of DukeScript, which is part of Apache NetBeans -- i.e., https://github.com/apache/incubator-netbeans-html4j

**Comment by Chris2011** [comment]
I mean a bugtracker for DukeScript, there is no bugtracker and if you mean, that I can use incubator-netbeans-html4j as bugtracking for DukeScript then ok, but this was no obvious. And first you have to activate the issue tracker on the repo so yeah I can't still create issues there.

In my opinion, I thought that DukeScript uses HTML4J but that doesn't mean that HTML4J is responsible for DukeScript problems per se. Only if we can figure out yeah that is a problem in HTML4J and not DukeScript then I should create a ticket there, otherwise I tried to find a bugtracker for DukeScript.

As I understand DukeScript based on HTML4J but it is NOT HTML4J or is NOT part of HTML4J. It is a whole different compoent. Please let me know when I'm wrong. I talked to Anton Epple, because he is the person, that I figured out is responsible for DukeScript, becaue of this page: https://dukescript.com/index.html#contact. No mentioning of Jaruslav etc. But this was not the case what I was asking for.

Let us discuss it in another thread for this, you can see my use case and you can see that the annotations will not fit in my case, because I don't want to write Java as much as I can. But if there is a better choice for me to use HTML4J, better than my implementation, then I will, but then I need help from the guys.

**Comment by eppleton** [comment]
Hi, DukeScript is based on html/java APIs. html/java is part of Apache NetBeans. When we were talking about your plugin, you were using plain html/Java. When using html/Java inside NetBeans, NetBeans issue tracker is the place to report issues. DukeScript.com uses html/Java independent of NetBeans and builds additional Presenters for iOS, Android, and more platforms.

But I admit that we're using this name "DukeScript" rather sloppily, because we're also the creators of the plugin which can be used for both plain html/Java and DukeScript Add-on Presenters so I understand the confusion. 

Toni (Anton Epple)

File: api.htmlui/test/unit/src/org/netbeans/modules/htmlui/AnnotationsTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/**
```
Comment (junichi11) [inline api.htmlui/test/unit/src/org/netbeans/modules/htmlui/AnnotationsTest.java]: Although it's a trivial thing, currently `/*` is used (#282).

Comment (jtulach) [reply api.htmlui/test/unit/src/org/netbeans/modules/htmlui/AnnotationsTest.java]: Thanks for spotting this - a copy&paste&merge error, I guess. Fixed in f59e878

**Comment by jtulach** [comment]
Is there any technical objections integrating these changes? If not, I'd like the change to be to merged.

**Comment by geertjanw** [comment]
OK, merging.

**Pull request merged**